src/app/clusters/groups-server/groups-server.cpp

```diff
@@ -232,141 +261,150 @@ bool emberAfGroupsClusterViewGroupCallback(app::CommandHandler * commandObj, con
+
+        err = commandObj->AddResponseData(commandPath, GetGroupMembershipResponse(commandData, iter));
+        iter->Release();
+        if (CHIP_NO_ERROR == err)
         {
-            app::CommandPathParams cmdParams = { emberAfCurrentEndpoint(), /* group id */ 0, Groups::Id,
-                                                 Commands::GetGroupMembershipResponse::Id,
-                                                 (app::CommandPathFlags::kEndpointIdValid) };
-            TLV::TLVWriter * writer          = nullptr;
-            SuccessOrExit(err = commandObj->PrepareCommand(cmdParams));
-            VerifyOrExit((writer = commandObj->GetCommandDataIBTLVWriter()) != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
-            SuccessOrExit(err = writer->Put(TLV::ContextTag(0), static_cast<uint8_t>(0xff)));
-            SuccessOrExit(err = writer->Put(TLV::ContextTag(1), count));
-            SuccessOrExit(err = writer->PutBytes(TLV::ContextTag(2), list, listLen));
-            SuccessOrExit(err = commandObj->FinishCommand());
+            status = EMBER_SUCCESS;
         }
     }
-    else
-    {
-        status = emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_NOT_FOUND);
-    }
-    if (EMBER_SUCCESS != status)
-    {
-        emberAfGroupsClusterPrintln("Groups: failed to send %s: 0x%x",
-                                    (groupCount == 0 || count != 0) ? "get_group_membership response" : "default_response", status);
-    }
 exit:
-    if (err != CHIP_NO_ERROR)
+    if (EMBER_SUCCESS != status)
     {
-        ChipLogError(Zcl, "Failed to encode response command.");
+        emberAfGroupsClusterPrint("Groups: failed to send get_group_membership response: 0x%x", status);
     }
     return true;
 }
 
 bool emberAfGroupsClusterRemoveGroupCallback(app::CommandHandler * commandObj, const app::ConcreteCommandPath & commandPath,
                                              const Commands::RemoveGroup::DecodableType & commandData)
 {
-    auto & groupId = commandData.groupId;
-
+    auto fabricIndex = GetFabricIndex(commandObj);
+    auto & groupId   = commandData.groupId;
+    auto endpointId  = commandPath.mEndpointId;
     EmberAfStatus status;
     CHIP_ERROR err = CHIP_NO_ERROR;
 
     emberAfGroupsClusterPrintln("RX: RemoveGroup 0x%2x", groupId);
 
-    status = removeEntryFromGroupTable(emberAfCurrentEndpoint(), groupId);
+    status = GroupRemove(fabricIndex, endpointId, groupId);
 
     // For all networks, Remove Group commands are only responded to when
     // they are addressed to a single device.
     if (emberAfCurrentCommand()->type != EMBER_INCOMING_UNICAST && emberAfCurrentCommand()->type != EMBER_INCOMING_UNICAST_REPLY)
     {
         return true;
     }
-
+#ifdef EMBER_AF_PLUGIN_SCENES
     // EMAPPFWKV2-1414: if we remove a group, we should remove any scene
     // associated with it. ZCL6: 3.6.2.3.5: "Note that if a group is
     // removed the scenes associated with that group SHOULD be removed."
```

Comment:
This seems like more legacy comments. Let's keep the quoted text (still in force), but the section number and the `EMAPPFWKV2-1414` are no longer contemporary.

Reply:
I have cleaned-up the comment.